jetty-util/src/main/java/org/eclipse/jetty/util/Pool.java

```diff
@@ -270,7 +296,7 @@ public Entry reserve(int allotment)
             if (space <= 0)
                 return null;
 
-            if (allotment >= 0 && (getReservedCount() * getMaxMultiplex()) >= allotment)
+            if (allotment >= 0 && getReservedCount() >= allotment)
```

Comment:
I would not touch the implementation of this method at all. We already had a discussion about the `allotment` parameter and concluded that it was a bad idea so we shifted away from this method that we only kept for backward compatibility. IMHO there's no point in modifying the implementation of something we know is fundamentally wrong at the high-level.